lib/rust/ensogl/component/flame-graph/src/mark.rs

```diff
@@ -0,0 +1,190 @@
+//! A single mark component that is used to build up a flame graph.
+
+use ensogl_core::display::shape::*;
+use ensogl_core::prelude::*;
+
+use ensogl::frp;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::shape::StyleWatchFrp;
+use ensogl_gui_component::component;
+use ensogl_gui_component::component::ComponentView;
+use ensogl_text as text;
+
+use super::BASE_TEXT_SIZE;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+
+const EMPTY_LABEL: &str = "<No Label>";
+
+const MARK_WIDTH: f32 = 2.0;
+const MARK_HOVER_AREA_WIDTH: f32 = 20.0;
+
+/// Invisible dummy color to catch hover events.
+const HOVER_COLOR: color::Rgba = color::Rgba::new(1.0, 0.0, 0.0, 0.000_001);
+
+const INFINITE: f32 = 99999.0;
+
+const TEXT_OFFSET_X: f32 = MARK_WIDTH + 8.0;
+
+
+
+// =========================
+// === Shape Definition ===
+// =========================
+
+mod background {
+    use super::*;
+    ensogl_core::define_shape_system! {
+        (style:Style) {
+            let width  : Var<Pixels> = MARK_WIDTH.px();
+            let height : Var<Pixels> = INFINITE.px();
+            let zoom                 = &Var::<f32>::from("1.0/zoom()");
+            let base_color           = style.get_color("flame_graph_mark_color");
+
+            let shape = Rect((&width*zoom,&height));
+            let shape = shape.fill(base_color);
+
+            let hover_area = shape.grow(zoom * MARK_HOVER_AREA_WIDTH.px());
+            let hover_area = hover_area.fill(HOVER_COLOR);
+
+
+            (shape + hover_area).into()
```

Comment:
please remove at least one empty line here. We can definitely use empty lines to group related lines together, but 2 empty lines is too much IMO. If you dont agree, lets talk about it! :)